server/helpers/index.js

```diff
@@ -1,6 +1,189 @@
+      })
+      return historicCompleted
+    })
+}
+
+const getCompletedKatas = (userName) => {
+  return new Promise((resolve, reject) => {
+    https.get(
+      `https://www.codewars.com/api/v1/users/${userName}/code-challenges/completed`,
+      (httpsResponse) => {
+        try {
+          if (httpsResponse.statusCode !== 200) {
+            throw new Error(`
+              Request Failed.
+              Status Code: ${httpsResponse.statusCode}
+              API: code-challenge.
+              User: ${userName}
+              Page: 0
+            `)
+          }
+        } catch (err) {
+          // Log err
+          reject()
+        }
+        let data = ''
+
+        httpsResponse.on('data', (chunk) => {
+          data += chunk
+        })
+
+        httpsResponse.on('end', () => {
+          try {
+            const response = JSON.parse(data)
+            if (response.success === false) {
+              throw new Error('User not found')
+            }
+
+            // Adds completed kata ids to an array
+            let completed = response.data.map(kata => kata.id)
+            if (response.totalPages === 1) {
+              resolve(completed)
+            } else {
+              // If there is more than one page of results all other pages must
+              // also be requested
+              getHistoricCompletedKatas(userName, response.totalPages)
+                .then((historicCompleted) => {
+                  completed = [...completed, ...historicCompleted]
+                  resolve(completed)
+                })
+                .catch((err) => {
+                  reject(err)
+                })
+            }
+          } catch (err) {
+            // If the user is not found it may indicate that their account has
+            // been deleted
+            // Log err
+            reject()
+          }
+        })
+      }
+    )
+  })
+}
+
+export const updateAllScores = async (katathonId) => {
+  try {
+    const katathon = await Katathon.findById(katathonId)
+    const { users, katas } = katathon
+    if (!users.length || !katas.length) {
+      return
+    }
+    // Compiles a list of all ongoing userScore requests. Once they have all
+    // updated and resolved the new data is written to db
+    const allPromises = []
+    users.forEach((user) => {
+      // This promise must always resolve so that if one user request fails the
+      // other scores will still be updated. Currently a deleted account will fail
+      const promise = new Promise(async (resolve) => {
+        try {
+          const completed = await getCompletedKatas(user.userName)
+
+          // This checks that no requests for completed katas have failed
+          if (!Array.isArray(completed)) {
+            throw new Error(`Failed to get completed katas for ${user.userName}`)
+          }
+
+          // Loops through katas and adds score when they are found in completed
+          let userScore = 0
+          katas.forEach((kata) => {
```

Comment:
try using .reduce() here instead of .forEach()

Reply:
Done